core/src/test/scala/caliban/execution/DefaultValueSpec.scala

```diff
@@ -0,0 +1,195 @@
+          )
+        },
+        testM("id validation") {
+          case class TestInput(@GQLDefault("1") id: UUID)
+          case class Query(test: TestInput => UUID)
+          val gql = graphQL(RootResolver(Query(i => i.id)))
+          assertM(gql.interpreter.run)(
+            fails(isSubtype[CalibanError.ValidationError](anything))
+          )
+        },
+        testM("boolean validation") {
+          case class TestInput(@GQLDefault("1") b: Boolean)
+          case class Query(test: TestInput => Boolean)
+          val gql = graphQL(RootResolver(Query(i => i.b)))
+          assertM(gql.interpreter.run)(
+            fails(isSubtype[CalibanError.ValidationError](anything))
+          )
+        },
+        testM("nullable validation") {
+          case class TestInput(@GQLDefault("1") s: Option[String])
+          case class Query(test: TestInput => String)
+          val gql = graphQL(RootResolver(Query(i => i.s.getOrElse("default"))))
+          assertM(gql.interpreter.run)(
+            fails(isSubtype[CalibanError.ValidationError](anything))
+          )
+        },
+        testM("successful nullable validation") {
+          case class TestInput(@GQLDefault("\"1\"") s: Option[String])
+          case class Query(test: TestInput => String)
+          val gql = graphQL(RootResolver(Query(i => i.s.getOrElse("default"))))
+          assertM(gql.interpreter)(anything)
+        },
+        testM("list validation") {
+          case class TestInput(@GQLDefault("\"string\"") string: List[String])
+          case class Query(test: TestInput => List[String])
+          val gql = graphQL(RootResolver(Query(i => i.string)))
+          assertM(gql.interpreter.run)(
+            fails(isSubtype[CalibanError.ValidationError](anything))
+          )
+        },
+        testM("successful list validation") {
+          case class TestInput(@GQLDefault("[\"string\"]") string: List[String])
+          case class Query(test: TestInput => List[String])
+          val gql = graphQL(RootResolver(Query(i => i.string)))
+          assertM(gql.interpreter)(anything)
+        },
+        testM("object validation") {
+          case class Nested(field: String)
+          case class TestInput(@GQLDefault("{field: 2}") nested: Nested)
+          case class Query(test: TestInput => String)
+          val gql = graphQL(RootResolver(Query(v => v.nested.field)))
+          assertM(gql.interpreter.run)(
+            fails(isSubtype[CalibanError.ValidationError](anything))
+          )
+        },
+        testM("successful object validation") {
+          case class Nested(field: String)
+          case class TestInput(@GQLDefault("{field: \"2\"}") nested: Nested)
+          case class Query(test: TestInput => String)
+          val gql = graphQL(RootResolver(Query(v => v.nested.field)))
+          assertM(gql.interpreter)(anything)
+        }
+        // NON_NULL
+      ),
+      testM("field default values") {
```

Comment:
A test that shows failed validation would be nice too.

Reply:
👍